cmd/template/dbdriver/files/service/redis.tmpl

```diff
@@ -0,0 +1,62 @@
+package database
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	_ "github.com/joho/godotenv/autoload"
+	"github.com/redis/go-redis/v9"
+)
+
+type Service interface {
+	Health() map[string]string
+}
+
+type service struct {
+	db *redis.Client
+}
+
+var (
+	address  = os.Getenv("DB_ADDRESS")
+	port     = os.Getenv("DB_PORT")
+	password = os.Getenv("DB_PASSWORD")
+	database = os.Getenv("DB_DATABASE")
+)
+
+func New() Service {
+	num, err := strconv.Atoi(database)
+	if err != nil {
+		log.Fatalf(fmt.Sprintf("database incorrect %v", err))
+	}
+
+	fullAddress := fmt.Sprintf("%s:%s", address, port)
+
+	rdb := redis.NewClient(&redis.Options{
+		Addr:     fullAddress,
+		Password: password,
+		DB:       num,
+	})
+
+	s := &service{db: rdb}
+
+	return s
+}
+
+func (s *service) Health() map[string]string {
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+
+	err := s.db.Ping(ctx)
+
+	if err != nil {
+		log.Fatalf(fmt.Sprintf("db down: %v", err))
+	}
+
+	return map[string]string{
+		"message": "It's healthy",
+	}
+}
```

Comment:
Ping is returning an error, and it is breaking the app when you curl the health route. 

I think this solves the issue

```go
func (s *service) Health() map[string]string {
	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
	defer cancel()

	result, err := s.db.Ping(ctx).Result()

	if err != nil {
		log.Fatalf(fmt.Sprintf("db down: %v", err))
	}

	 if result != "PONG" {
        log.Fatalf("Unexpected ping response: %s", result)
    }

	return map[string]string{
		"message": "It's healthy",
	}
}
```

Reply:
Oh good catch 👍